src/heimdall-tree/index.js

```diff
@@ -50,10 +50,11 @@ function statsFromCounters(counterStore, counterCache) {
 }
 
 export default class HeimdallTree {
-  constructor(heimdall) {
+  constructor(heimdall, lastKnownTime) {
     this._heimdall = heimdall;
     this.root = null;
     this.format = heimdall && heimdall._timeFormat ? heimdall._timeFormat : format;
+    this.lastKnownTime = lastKnownTime ? normalizeTime(lastKnownTime) : now();
```

Comment:
the "or" logic here should actually happen within `tree.construct` as it's possible to call construct many times on an active heimdall tree if paired with a heimdall instance at runtime.